packages/core/src/StarknetWindowObject.ts

```diff
@@ -18,44 +75,157 @@ export type WalletEvents =
+     */
+    abi?: string
+  }
+}
+export interface AddDeclareTransactionResult {
+  /**
+   * The hash of the declare transaction
+   */
+  transaction_hash: FELT
+  /**
+   * The hash of the declared class
+   */
+  class_hash: FELT
+}
+
+/**
+ * DEPLOY_ACCOUNT_TXN_V1
+ * @see https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json
+ */
+export interface AddDeployAccountTransactionParameters {
+  /**
+   * The salt for the address of the deployed contract
+   */
+  contract_address_salt: FELT
+  /**
+   * The parameters passed to the constructor
+   */
+  constructor_calldata: FELT[]
+  /**
+   * The hash of the deployed contract's class
+   */
+  class_hash: FELT
+}
+export interface AddDeployAccountTransactionResult {
+  /**
+   * The hash of the deploy transaction
+   */
+  transaction_hash: FELT
+  /**
+   * The address of the new contract
+   */
+  contract_address: FELT
+}
+
+/**
+ * EIP-1102:
+ * @see https://eips.ethereum.org/EIPS/eip-1102
+ */
+export interface RequestAccountsParameters {
+  /**
+   * If true, the wallet will not show the wallet-unlock UI in case of a locked wallet,
+   * nor the dApp-approve UI in case of a non-allowed dApp.
+   */
+  silentMode?: boolean
+}
+
+/**
+ * EIP-747:
+ * @see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-747.md
+ */
 export interface WatchAssetParameters {
   type: "ERC20" // The asset's interface, e.g. 'ERC20'
   options: {
-    address: string // The hexadecimal StarkNet address of the token contract
+    address: string // The hexadecimal Starknet address of the token contract
     symbol?: string // A ticker symbol or shorthand, up to 5 alphanumerical characters
     decimals?: number // The number of asset decimals
     image?: string // A string url of the token logo
     name?: string // The name of the token - not in spec
   }
 }
 
-// EIP-3085
-// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3085.md
-
+/**
+ * EIP-3085:
+ * @see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3085.md
+ */
 export interface AddStarknetChainParameters {
   id: string
   chainId: string // A 0x-prefixed hexadecimal string
   chainName: string
-  baseUrl: string
   rpcUrls?: string[]
   blockExplorerUrls?: string[]
 
   nativeCurrency?: {
-    address: string // Not part of the standard, but required by StarkNet as it can work with any ERC20 token as the fee token
+    address: string // Not part of the standard, but required by Starknet as it can work with any ERC20 token as the fee token
     name: string
     symbol: string // 2-6 characters long
     decimals: number
   } // Currently ignored.
   iconUrls?: string[] // Currently ignored.
 }
 
-export interface SwitchStarknetChainParameter {
+export interface SwitchStarknetChainParameters {
   chainId: string // A 0x-prefixed hexadecimal string
 }
 
 export type RpcMessage =
+  | {
+      type: "wallet_getPermissions"
+      params: never
```

Comment:
@avimak 
There is no way to define that type of message. In order to support a message without params, just omit it:
```
{
      type: "wallet_getPermissions"
       result: Permission[]
}
```

https://www.typescriptlang.org/play#code/C4TwDgpgBAglC8UDeAoK6oEMBcUDOwATgJYB2A5gDRoYBGupEAbhIQNw3oDGuBJFHAL4oUXAPakCWbHESoM0qAHIAJkuoKeq9Zyj0ojFoRSCgA